include/cetl/unbounded_variant.hpp

```diff
@@ -48,36 +48,461 @@ class bad_unbounded_variant_access : public std::bad_cast
 #endif  // defined(__cpp_exceptions) || defined(CETL_DOXYGEN)
 
 // Forward declarations
-template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment>
+template <std::size_t Footprint, bool Copyable, bool Movable, std::size_t Alignment, bool IsPmr>
 class unbounded_variant;
 
 namespace detail
 {
 
+inline std::nullptr_t throw_bad_alloc()
+{
+#if defined(__cpp_exceptions)
+    throw std::bad_alloc();
+#else
+    return nullptr;
```

Comment:
We don't terminate here but instead return `nullptr` which makes unbounded variant as "valueless by exception".